locked-nft/contracts/LockedNFT.cdc

```diff
@@ -0,0 +1,154 @@
+import NonFungibleToken from "./NonFungibleToken.cdc"
+
+pub contract LockedNFT {
+
+    pub event ContractInitialized()
+    pub event NFTLocked(
+        id: UInt64,
+        to: Address?,
+        lockedAt: UInt64,
+        lockedUntil: UInt64,
+        duration: UInt64
+    )
+    pub event NFTUnlocked(
+        id: UInt64,
+        from: Address?
+    )
+
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+
+    pub var totalLockedTokens:      UInt64
+    access(self) let lockedTokens:  {String: LockedData}
+
+    pub struct LockedData {
+        pub let id: UInt64
+        pub let owner: Address
+        pub let lockedAt: UInt64
+        pub let lockedUntil: UInt64
+        pub let duration: UInt64
+        pub let nftType: String
+
+        init (id: UInt64, owner: Address, duration: UInt64, nftType: String) {
```

Comment:
Can NFT Type be of type instead of string?

Reply:
Yes. That's better. Have switched this to a type.